nb-adapters/adapter-s4j/src/main/java/io/nosqlbench/adapter/s4j/S4JSpace.java

```diff
@@ -217,45 +221,44 @@ public void incTxnBatchTrackingCnt() {
-
-                    JMSContext jmsConnContext = getOrCreateConnLvlJMSContext(s4jConnFactory, s4JClientConnInfo, sessionMode);
-                    jmsConnContext.setClientID(connLvlJmsConnContextIdStr);
-                    jmsConnContext.setExceptionListener(e -> {
-                        if (logger.isDebugEnabled()) {
-                            logger.error("onException::Unexpected JMS error happened:" + e);
-                        }
-                    });
-
-                    connLvlJmsContexts.put(connLvlJmsConnContextIdStr, jmsConnContext);
+        Map<String, Object> cfgMap;
+        try {
+            cfgMap = s4JClientConnInfo.getS4jConfObjMap();
+
+            for (int i=0; i<getMaxNumConn(); i++) {
+                String connLvlJmsConnContextIdStr =getConnLvlJmsContextIdentifier(i);
+
+                // Establish a JMS connection
+                PulsarConnectionFactory s4jConnFactory = connFactories.computeIfAbsent(
+                    connLvlJmsConnContextIdStr,
+                    __ -> new PulsarConnectionFactory(cfgMap));
+
+                JMSContext jmsConnContext = getOrCreateConnLvlJMSContext(s4jConnFactory, s4JClientConnInfo, sessionMode);
+                jmsConnContext.setClientID(connLvlJmsConnContextIdStr);
+                jmsConnContext.setExceptionListener(e -> {
                     if (logger.isDebugEnabled()) {
-                        logger.debug("[Connection level JMSContext] {} -- {}",
-                            Thread.currentThread().getName(),
-                            jmsConnContext );
+                        logger.error("onException::Unexpected JMS error happened:" + e);
                     }
+                });
+
+                connLvlJmsContexts.put(connLvlJmsConnContextIdStr, jmsConnContext);
+                if (logger.isDebugEnabled()) {
+                    logger.debug("[Connection level JMSContext] {} -- {}",
+                        Thread.currentThread().getName(),
+                        jmsConnContext );
                 }
             }
-            catch (JMSRuntimeException e) {
-                logger.error("Unable to initialize JMS connection factory with the following configuration parameters: {}", s4JClientConnInfo.toString());
-                throw new S4JAdapterUnexpectedException("Unable to initialize JMS connection factory with the following error message: " + e.getCause());
-            }
-            catch (Exception e) {
-                e.printStackTrace();
-            }
+        }
+        catch (JMSRuntimeException e) {
+            logger.error("Unable to initialize JMS connection factory with the following configuration parameters: {}", s4JClientConnInfo.toString());
+            throw new S4JAdapterUnexpectedException("Unable to initialize JMS connection factory with the following error message: " + e.getCause());
+        }
+        catch (Exception e) {
+            e.printStackTrace();
```

Comment:
Exceptions should not be swallowed here. If the intent is to ignore them during testing, then the error handler stack should be employed to do this as with `errors=MyErrorTypePattern.*:ignore,...`

Reply:
Thanks, this block is really not necessary and removed it.